File: src/main/java/org/example/controller/ApiController.java
package org.example.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping(path = "/api")
public class ApiController {

    @GetMapping(path = "version")
    public Map<String, String> version() {
        return Map.of("version", "1.2.3");
    }

    @GetMapping(path = "status")
    public Map<String, String> status() {
        return Map.of("status", "UP");
    }

}

File: src/main/kotlin/org/example/controller/ApiController.kt
package org.example.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class ApiController {

    @GetMapping("version")
    fun version(): Map<String, String> {
        return mapOf("version" to "1.2.3")
    }

    @GetMapping("status")
    fun status(): Map<String, String> {
        return mapOf("status" to "UP")
    }
}
